profiling/src/lib.rs

```diff
@@ -1071,6 +1099,60 @@ extern "C" fn execute_internal(
     interrupt_function(execute_data);
 }
 
+/// Find out the reason for the current garbage collection cycle. If there is
+/// a `gc_collect_cycles` function at the top of the call stack, it is because
+/// of a userland call  to `gc_collect_cycles()`, otherwise the engine decided
+/// to run it.
+#[cfg(feature = "timeline")]
+unsafe fn gc_reason() -> &'static str {
+    let execute_data = zend::ddog_php_prof_get_current_execute_data();
+    let fname = || execute_data.as_ref()?.func.as_ref()?.name();
+    match fname() {
+        Some(name) if name == b"gc_collect_cycles" => "induced",
+        _ => "engine",
+    }
+}
```

Comment:
@realFlowControl I refactored this bit, partly because I wasn't sure func would always be there and so I didn't trust the pointer dereferencing.